File: libs/interfaces/history.ts
import { Exercise } from './exercise';
import { User } from './user';
import { Workout } from './workout';

export interface History {
    id?: number;
    workout?: Workout;
    user?: User;
    exercise?: Exercise;
    weight?: number;
    reps?: number;
    restTime?: string;
    date?: Date | string,
    groupedHistory?: GroupedHistory[]
}


export interface GroupedHistory {
    id: number;
    weight: number;
    reps: number;
}

File: my-workout-api/src/main/java/com/avernet/my_workout_api/History/History.java
package com.avernet.my_workout_api.History;

import com.avernet.my_workout_api.Workout.Workout;
import com.avernet.my_workout_api.exercise.Exercise;

public record History(
    Long id,
    Exercise exercise,
    Float weight,
    Short reps,
    Workout workout
) {
}
